src/ILVerify/tests/ILMethodTester.cs

```diff
@@ -0,0 +1,210 @@
+        /// </summary>      
+        public static IEnumerable<Object[]> GetMethodsWithInvalidIL()
+        {
+            var methodSelector = new Func<string[], EcmaMethod, EcmaMethodIL, TestCase>((mparams, method, methodIL) =>
+            {
+                if (mparams.Length == 4 && mparams[1].ToLower() == "invalid")
+                {
+                    var expectedErrors = mparams[2].Split('.');
+                    List<VerifierError> verificationErros = new List<VerifierError>();
+
+                    foreach (var item in expectedErrors)
+                    {
+                        if (Enum.TryParse(item, out VerifierError expectedError))
+                        {
+                            verificationErros.Add(expectedError);
+                        }
+                    }
+
+                    var newItem = new InvalidILTestCase { ILImporter = new ILImporter(method, methodIL) };
+
+                    if (expectedErrors.Length > 0)
+                    {
+                        newItem.ExpectedVerifierError = verificationErros;
+                    }
+
+                    if (mparams[3].Length > 0)
+                    {
+                        newItem.ExpectedException = Type.GetType(mparams[3]);
+
+                        if (newItem.ExpectedException == null)
+                        {
+                            var ilVerifyAssembly = System.Reflection.Assembly.GetAssembly(typeof(VerifierError));
+                            newItem.ExpectedException = ilVerifyAssembly.GetType(mparams[3]);
+                        }
+                    }
+
+                    newItem.MethodName = method.Name;
+                    return newItem;
+                }
+                return null;
+            });
+            return GetTestMethodsFromDll(methodSelector);
+        }
+
+        private static IEnumerable<Object[]> GetTestMethodsFromDll(Func<string[], EcmaMethod, EcmaMethodIL, TestCase> MethodSelector)
+        {
+            List<TestCase[]> retVal = new List<TestCase[]>();
+
+            foreach (var testDllName in GetAllTestDlls())
+            {
+                var testModule = TestDataLoader.GetModuleForTestAssembly(testDllName);
+
+                foreach (var methodHandle in testModule.MetadataReader.MethodDefinitions)
+                {
+                    var method = (EcmaMethod)testModule.GetMethod(methodHandle);
+                    var methodIL = EcmaMethodIL.Create(method);
+                    if (methodIL == null)
+                        continue;
+
+                    var methodName = method.ToString();
+
+                    if (!String.IsNullOrEmpty(methodName) && methodName.Contains("_"))
+                    {
+                        var mparams = methodName.Split('_');
+                        var newItem = MethodSelector(mparams, method, methodIL);
+
+                        if (newItem != null)
+                        {
+                            retVal.Add(new TestCase[] { newItem });
```

Comment:
Is there something better than this? According to what I found the method feeding `MemberData `must return `IEnumerable<Object[]>`, but what we need is just` IEnumrable<TestCase>`